### PR TITLE
ui/services: convert IntegrationKeyCreateDialog & IntegrationKeyList  to ts

### PR DESCRIPTION
**Pull request by tony-tvu**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Part of an ongoing effort to convert existing javascript files to typescript.

**Note:** IntegrationKeyCreateDialog and IntegrationKeyList were both converted in this PR to avoid receiving this error: 
`Cannot destructure property 'service' of 'cache5.readQuery(...)' as it is null.`
This error occurs when IntegrationKeyCreateDialog.js created a new integration key without a cache available in IntegrationKeyList.tsx, since IntegrationKeyList.tsx uses urql instead of apollo now.

**Which issue(s) this PR fixes:**
https://github.com/target/goalert/issues/2318

**Out of Scope:**
- switch out Apollo client with urql

### CODE REVIEW

File: web/src/app/services/IntegrationKeyList.tsx [hunk 1]
```diff
@@ -71,27 +75,24 @@ export function IntegrationKeyDetails(props) {
       {props.type === 'email' && (
         <RequireConfig
           configID='Mailgun.Enable'
-          else='Email integration keys are currently disabled.'
+          else={<div>Email integration keys are currently disabled.</div>}
```
Comment (KatieMSB) [inline web/src/app/services/IntegrationKeyList.tsx:78]:
```suggestion
          else={
            <React.Fragment>
              Email integration keys are currently disabled.
            </React.Fragment>
          }
```
Looks like the `<div>` here is throwing an error

Screen Shot 2022-09-08 at 9 44 22 AM

**Review by KatieMSB — CHANGES_REQUESTED** [review]

**Review by KatieMSB — DISMISSED** [review]

**Review by KatieMSB — DISMISSED** [review]

File: web/src/app/services/IntegrationKeyCreateDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export default function IntegrationKeyCreateDialog(props: {
+  serviceID: string
+  onClose: () => void
+}): JSX.Element {
+  const [value, setValue] = useState<Value | null>(null)
+  const { serviceID, onClose } = props
+
+  const [createIntegrationKeyStatus, createIntegrationKey] =
+    useMutation(mutation)
```
Comment (Forfold) [inline web/src/app/services/IntegrationKeyCreateDialog.tsx:28]: just a suggestion, might read easier to shorten the names here

```suggestion
  const [createKeyStatus, createKey] =
    useMutation(mutation)
```

File: web/src/app/services/IntegrationKeyList.tsx [hunk 3]
```diff
@@ -17,6 +16,13 @@ import AppLink from '../util/AppLink'
 import makeStyles from '@mui/styles/makeStyles'
 import Spinner from '../loading/components/Spinner'
 import { GenericError } from '../error-pages'
+import { IntegrationKey } from '../../schema'
+
+interface Item {
```
Comment (Forfold) [inline web/src/app/services/IntegrationKeyList.tsx:21]: let's import and use the `FlatListListItem` type instead

**Review by Forfold — CHANGES_REQUESTED** [review]
just a couple small things ^ otherwise looks good

**Review by Forfold — DISMISSED** [review]

**Review by KatieMSB — APPROVED** [review]

**Review by Forfold — APPROVED** [review]

**Pull request merged**